consume fully configurable

rajatagarwal-ibm opened the pull request:
### Description

<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

ocofaigh commented on stack_definition.json:
```diff
@@ -392,7 +486,7 @@
       ],
       "name": "4b - Secrets Manager",
-      "version_locator": "7a4d68b4-cf8b-40cd-a3d1-f49aff526eb3.db823650-3010-4482-b807-45145f273553-global"
+      "version_locator": "7a4d68b4-cf8b-40cd-a3d1-f49aff526eb3.e204ceb9-6339-4c6f-b627-c648ae45cd97-global"
```
won't you need to remove the `iam_engine_enabled` input - its not a valid input for the new variation

ocofaigh reviewed (changes requested):
I think we should wait until App config has KMS and Event Notification support (Mukul is working on adding this now). After that, the dependency tree would look like this:

1a - Key management
1b - Object storage
1c - Cloud Monitoring
2 - Event Notifications
3a - Cloud Logs for logging
3b - Cloud Logs for activity tracking
3c - App Config
3d - Secrets Manager
4a - Security and Compliance Center Workload Protection
4b - Activity Tracker Event Routing

Renaming stack members is currently a breaking change, however since this whole update is a breaking change anyway, we can use this opportunity to rename. We don't have the ability to order stack members either, so using numbers is the only way currently (I'l follow up with projects to see if we can get this ability any time soon).

I'm also wondering whether it makes sense to actually rename the variation programmatic name to prevent users from being able to upgrade, but that will all depend on what upgrade experience is - lets jump on a call to go through that before we decide.

rajatagarwal-ibm commented:
Update the diagram
	- App config arrow to EN and KMS
	- 2 more buckets: AT Cloud logs bucket, AT Cloud logs Metrics Bucket

ocofaigh commented on reference-architectures/reference-architecture-css.md:
```diff
@@ -1,8 +1,8 @@
 copyright:
-  years: 2024
-lastupdated: "2024-12-05"
+  years: 2025
```
I think you are supposed to keep 2024
```suggestion
  years: 2024, 2025
```

ocofaigh commented on stack_definition.json:
```diff
@@ -179,39 +179,69 @@
+          "value": "ref:../../inputs/existing_resource_group_name"
+        },
+        {
+          "name": "use_existing_resource_group",
```
hmm, you have this same issue in KMS, COS and Secrets Manager

ocofaigh reviewed (changes requested):
Rajat can you please do a full review of all the inputs you are using for the DAs. Some of the inputs are not even valid inputs for the new variations. 
Also some of the inputs can be removed, or updated. Things like `secrets_manager_instance_name` where we hard coded to "base-security-services-sm" to prevent a breaking change. We can reset all these now as this is a major version release with no upgrade path anyway

ocofaigh commented:
Lets review the output names too. for example `en_crn` -> `event_notifications_crn`

rajatagarwal-ibm commented:
Thanks @ocofaigh for the review, I will go thru all the inputs and outpus for all the variations and mapping too.

rajatagarwal-ibm commented:
@ocofaigh, I have updated the diagram and fixed crooked lines and added a bucket for AT. However, I don't fully understand the second cloud logs bucket requirement. Perhaps we can have a quick discussion on Monday AM.

rajatagarwal-ibm commented:
@ocofaigh addressed all the comments. Also tested the new fix from Projects and added back `existing_event_notifications_instances` mapping for both cloud logs instances.

ocofaigh commented on ibm_catalog.json:
```diff
@@ -256,16 +258,52 @@
+              "key": "key_protect_plan",
+              "type": "string",
+              "default_value": "tiered-pricing",
+              "description": "The service plan of the Key Protect instance that will be provisioned by this solution. Only used if not supplying `existing_kms_instance_crn`. Learn more: https://cloud.ibm.com/docs/key-protect?topic=key-protect-pricing-plan",
```
Use a symlink for the learn more url

ocofaigh commented on ibm_catalog.json:
```diff
@@ -256,16 +258,52 @@
+                },
+                {
+                  "displayname": "cross-region-resiliency",
+                  "value": "cross-region-resiliency"
```
Can we give these better display names - like they show in the service tile UI?

ocofaigh commented on ibm_catalog.json:
```diff
@@ -256,16 +258,52 @@
+                },
+                {
+                  "displayname": "graduated-tier",
+                  "value": "graduated-tier"
```
Can we give these better display names - like they show in the service tile UI?

ocofaigh reviewed (changes requested):
Diagram updates:
- Can we keep the data commincation lines seperate instead of combinibng them - its very hard to know which services talk to which other services
- Arrow needs to be switched for SCC WP and App Config
- `App configuration` -> `App Configuration`
- There is no integration between SCC-WP and Cloud Monitoring - remove the line
- All of the buckets are configured with Cloud Monitoring - there should be a line to show this
- Rename `Activity Tracker bucket` and `Activity Tracker Metrics bucket` to `Cloud Logs for AT data bucket` and `Cloud Logs for AT metrics bucket`
- Rename `Cloud Logs bucket` and `Cloud Logs Metrics` bucket to `Cloud Logs data bucket` and `Cloud Logs metrics bucket`

ocofaigh commented:
/run pipeline

ocofaigh commented on ibm_catalog.json:
```diff
@@ -71,7 +71,9 @@
       "flavors": [
         {
           "label": "Standard",
```
Lets come up with a better variation name here. I would like to use a different label name to the current variation since there actually won't be any migration path between them. Perhaps "Click and go" ?

ocofaigh commented on ibm_catalog.json:
```diff
@@ -256,16 +258,52 @@
+              "key": "key_protect_plan",
+              "type": "string",
+              "default_value": "tiered-pricing",
+              "description": "The service plan of the Key Protect instance that will be provisioned by this solution. Only used if not supplying `existing_kms_instance_crn`. Learn more [here](https://cloud.ibm.com/docs/key-protect?topic=key-protect-pricing-plan).",
```
For consistency with other DAs
```suggestion
              "description": "The service plan of the Key Protect instance that will be provisioned by this solution. Only used if not supplying `existing_kms_instance_crn`. [Learn more](https://cloud.ibm.com/docs/key-protect?topic=key-protect-pricing-plan).",
```

ocofaigh commented on ibm_catalog.json:
```diff
@@ -278,19 +278,19 @@
                 },
                 {
-                  "displayname": "graduated-tier-sysdig-secure-plus-monitor (Available in eu-fr2 region only.)",
+                  "displayname": "Graduated Tier - Sysdig Secure + Monitor (Available in eu-fr2 region only.)",
```
ah lets remove this if its only supported in this eu-fr2

rajatagarwal-ibm commented:
/run pipeline

rajatagarwal-ibm commented:
/run pipeline

rajatagarwal-ibm commented:
/run pipeline

ocofaigh commented on ibm_catalog.json:
```diff
@@ -70,8 +70,10 @@
-          "label": "Standard",
-          "name": "standard",
+          "label": "Click and go",
+          "name": "standard-fully-config",
```
maybe we go with `click-and-go` to match the label? Also don't forget this also needs to be updated in `.catalog-onboard-pipeline.yaml`

ocofaigh commented on stack_definition.json:
```diff
@@ -402,27 +534,27 @@
       "value": "ref:../../members/1b - Object storage/outputs/cos_instance_id"
     },
     {
       "name": "en_crn",
```
```suggestion
      "name": "event_notifications_crn",
```

ocofaigh reviewed (changes requested):
final few comments

ocofaigh commented:
It says `(1a - Key management) Validation result: failed` however the workspace passed:
<img width="776" height="846" alt="image" src="https://github.com/user-attachments/assets/cc76b31f-d5e0-43d2-9b4b-c7329e9c8cba" />
